cg/services/sample_service/utils.py

```diff
@@ -0,0 +1,20 @@
+from datetime import datetime
+
+
+def get_cancel_comment(user_name: str) -> str:
+    date: str = datetime.now().strftime("%Y-%m-%d")
+    return f"Cancelled {date} by {user_name}"
+
+
+def get_confirmation_message(sample_ids: list[str], remaining_cases: list[str]) -> str:
+    samples = "sample" if len(sample_ids) == 1 else "samples"
+    message = f"Cancelled {len(sample_ids)} {samples}. "
+
+    if remaining_cases:
+        cases = "case" if len(remaining_cases) == 1 else "cases"
+        case_message = ", ".join(remaining_cases)
+        message += f"Found {len(remaining_cases)} {cases} with additional samples: {case_message}."
+    else:
+        message += "No case contained additional samples."
+
+    return message
```

Comment:
This method is very hard to follow. I think it stems from 

1. The `samples` and `cases` being instantiated in a convoluted way. In general it is quite dense with if/elses.
2. `remaining_cases` is a perplexing parameter name. From the context it is "cases with more samples than the one being cancelled" but my first thought would be that it is "cases that have yet to be handled in this flow".
3. The string `f"Found {len(remaining_cases)} {cases} with additional samples: {case_message}."`I would most likely read expecting `case_message` to be the list of additional samples.

Reply:
Made more readable in https://github.com/Clinical-Genomics/cg/pull/3507/commits/53e15193a73f78c51c9120e88cf7bb96c961ea6b

Feel free to suggest further improvements 🙂